br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -504,8 +534,143 @@ func (em *ErrorManager) ResolveAllConflictKeys(
 	return errors.Trace(g.Wait())
 }
 
+// ReplaceConflictKeys query all conflicting rows (handle and their
+// values) from the current error report and resolve them concurrently
+// by replacing the necessary rows and reserving the others.
+func (em *ErrorManager) ReplaceConflictKeys(
+	ctx context.Context,
+	tbl tidbtbl.Table,
+	tableName string,
+	pool *utils.WorkerPool,
+	decoder *kv.TableKVDecoder,
+	fnGetLatest func(ctx context.Context, key []byte) ([]byte, error),
+	fnDeleteKey func(ctx context.Context, key []byte) error,
+) error {
+	if em.db == nil {
+		return nil
+	}
+
+	sessionOpts := encode.SessionOptions{
+		// TODO: need to find the correct value for SQLMode
+		SQLMode: mysql.ModeStrictAllTables,
+	}
+	encoder, err := kv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+		Logger:         log.L(),
+	})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	g, gCtx := errgroup.WithContext(ctx)
+	pool.ApplyOnErrorGroup(g, func() error {
+		// check index KV first
+		rawHandleRows, err := em.db.QueryContext(
+			gCtx, fmt.Sprintf(selectIndexConflictKeysReplace, em.schemaEscaped),
+			tableName)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		for rawHandleRows.Next() {
+			var rawKey []byte
+			if err := rawHandleRows.Scan(&rawKey); err != nil {
+				return errors.Trace(err)
+			}
+			em.logger.Debug("got rawKey from table",
+				zap.ByteString("rawKey", rawKey))
+
+			var value []byte
+			value, err = fnGetLatest(gCtx, rawKey)
```

Comment:
```suggestion
			value, err := fnGetLatest(gCtx, rawKey)
```

generally we don't need an extra line to define a new variable when it's a return value. Same for below codes